### PR TITLE
FEATURE: ask for confirmation before discarding

### PR DESCRIPTION
**Pull request by dimaip**
Resolves: #1387

It prompts for discard confirmation both for discard and discard all.

image

### CODE REVIEW

**Comment by dimaip** [comment]
Damn, need to adjust the tests, will do this evening.

File: packages/neos-ui/src/Containers/Modals/DiscardDialog/index.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            <Dialog
+                actions={[this.renderAbort(), this.renderConfirm()]}
+                title={this.renderTitle()}
+                onRequestClose={this.handleAbort}
+                isOpen
+                >
+                <div className={style.modalContents}>
+                    <I18n
+                        id="Neos.Neos:Main:content.components.discardAllDialog.discardXChangesSubheader"
+                        params={{numberOfChanges: nodesToBeDiscarded.count()}}
```
Comment (mstruebing) [inline packages/neos-ui/src/Containers/Modals/DiscardDialog/index.js]: A fallback would be nice :)

Comment (mstruebing) [reply packages/neos-ui/src/Containers/Modals/DiscardDialog/index.js]: Wouldn't something like:
```
<i18n
    id="Neos.Neos:Main:content.components.discardAllDialog.discardXChangesSubheader"
    params={{numberOfChanges: nodesToBeDiscarded.count()}}
    fallback=`myFallBackString nodesThatWillBeDiscarded: ${nodesToBeDiscarded.count()}`
```

possible?

Comment (dimaip) [reply packages/neos-ui/src/Containers/Modals/DiscardDialog/index.js]: Ah you are genius!

File: packages/neos-ui-redux-store/src/CR/Workspaces/index.js [hunk 2]
```diff
@@ -9,13 +9,17 @@ import * as selectors from './selectors';
 
 const UPDATE = '@neos/neos-ui/CR/Workspaces/UPDATE';
 const PUBLISH = '@neos/neos-ui/CR/Workspaces/PUBLISH';
-const DISCARD = '@neos/neos-ui/CR/Workspaces/DISCARD';
+const COMMENCE_DISCARD = '@neos/neos-ui/CR/Workspaces/COMMENCE_DISCARD';
```
Comment (dfeyer) [inline packages/neos-ui-redux-store/src/CR/Workspaces/index.js]: Nitpick or maybe by french knowledge ... Commence is valid english, but unusual in this context why not "BEGIN_DISCARD" ...

Comment (dimaip) [reply packages/neos-ui-redux-store/src/CR/Workspaces/index.js]: Totally agree, but for the sake of consistency with other sagas where we use commence, e.g. delete node saga.

**Comment by dimaip** [comment]
Ok, tests green, feedback addressed, merging based on previous reviews.

**Pull request merged**